exporter/awskinesisexporter/internal/batch/batch.go

```diff
@@ -42,7 +43,9 @@ type Batch struct {
 
 	compression compress.Compressor
 
-	records []*kinesis.PutRecordsRequestEntry
+	records        []*kinesis.PutRecordsRequestEntry
+	useAggregation bool
+	aggregator     *Aggregator
```

Comment:
I would typically like having a `NoOpAggregator` so that way it doesn't require continual checks `if thing exist, do extra code path`.

That way the cost in using the Batch type is fixed (mostly) and isn't drastically changed when branch optimisation happens.